src/sentry/auth/helper.py

```diff
@@ -466,6 +472,8 @@ def handle_unknown_identity(
                 else:
                     # assume they've confirmed they want to attach the identity
                     op = "confirm"
+            elif is_account_verified:
+                op = "confirm"
```

Comment:
This seems like it could be at risk of "an attempt to hijack membership of another organization" as mentioned in the comment above. I invite careful review to make sure that's not the case.

My solution is predicated on:

1. By the time the initial login flow reaches `AccountConfirmLink.store_in_redis`, the user has been authenticated by the org's auth provider, which suffices to allow org membership. (We would proceed to `handle_attach_identity` if the Sentry user were authenticated at that time.)
2. The check on `get_verification_value_from_key` suffices to prove that this is the same user who was authenticated in that way.

Per discussion earlier today, it's possible this is an attacker who has gained access to the real user's inbox. But if the attacker can bypass the SSO check *and* could reset that user's Sentry password, they'd get access regardless.